transformers.py

```diff
@@ -0,0 +1,151 @@
+import sonnet as snt
+import tensorflow as tf
+import math
+
+def positional_encoding(seq_len, d_model, batch_size=1):
+    """
+    Returns a tensor following the postional encoding function
+     (sinusodal from Vaswani et. all 2017).
+
+    Return shape: [batch, seq_len, d_model]
+    """
+    def encoding_angle(pos, i):
+        pos = tf.cast(pos, tf.dtypes.float32)
+        i = tf.cast(i, tf.dtypes.float32)
+        denom = tf.math.pow(10000, i/d_model)
```

Comment:
The exponent should be `2*i/d_model`

Reply:
I used this:
```
        d = tf.cast(d_model, tf.dtypes.float32)
        denom = tf.math.pow(10000., 2. * i / d)
```

Reply:
![image](https://user-images.githubusercontent.com/3289054/128637853-6b1803c2-0d50-4206-adad-2886882f53b6.png)

I had some trouble following this, since the LHS has an angle of `2i` i follow why they use `2i` in the RHS, but in my case I'm only using `i` so shouldn't it only be `i`?

Reply:
No, you still want `2i`.